### ISSUE TITLE
signals after caching logits

### ISSUE THREAD

**Issue by fazelehh** [aidotse/LeakPro#339]
# Issue
## Problem Description
After caching logits of shadow models, signal.py, signal_extractor-py and all attacks' signals should be updated. So far, we have BASE, LiRA, and RMIA in place.
